lib/puppet/provider/package/dnfmodule.rb

```diff
@@ -34,10 +34,11 @@ def self.prefetch(packages)
 
   def self.instances
     packages = []
-    cmd = "#{command(:dnf)} module list --enabled -d 0 -e #{error_level}"
+    cmd = "#{command(:dnf)} module list --enabled -d 0 -e #{error_level};"\
+          "#{command(:dnf)} module list --disabled -d 0 -e #{error_level}"
```

Comment:
I would call `dnf module list` without `--enabled` or `--disabled` here so we get all modules in one execution.